torch/quantization/quantize_fx.py

```diff
@@ -0,0 +1,149 @@
+from .fx import Fuser  # noqa: F401
+from .fx import Quantizer  # noqa: F401
+
+def fuse_fx(graph_module, inplace=False):
+    fuser = Fuser()
+    return fuser.fuse(graph_module, inplace)
+
+def _prepare_fx(graph_module, qconfig_dict, inplace, is_dynamic):
+    quantizer = Quantizer()
+    prepare = quantizer.prepare_dynamic if is_dynamic else quantizer.prepare
+    prepared = prepare(graph_module, qconfig_dict, inplace)
+    return prepared
+
+def prepare_fx(graph_module, qconfig_dict, inplace=False):
+    """ If graph_module is in training mode, the model will be
+    prepared as a qat model, otherwise, it will be prepared as a
+    model used for post training static quantization.
+
+    Args:
+      graph_module: model from symbolic_tracing (torch.fx)
+      qconfig_dict: tbd
+    """
+    return _prepare_fx(graph_module, qconfig_dict, inplace, False)
+
+def prepare_dynamic_fx(graph_module, qconfig_dict, inplace=False):
+    return _prepare_fx(graph_module, qconfig_dict, inplace, True)
+
+def _convert_fx(graph_module, inplace=False, debug=False, is_dynamic=False):
+    quantizer = Quantizer()
+    return quantizer.convert(graph_module, inplace, debug, is_dynamic)
+
+def convert_fx(graph_module, inplace=False, debug=False):
+    return _convert_fx(graph_module, inplace, debug, is_dynamic=False)
+
+def convert_dynamic_fx(graph_module, inplace=False, debug=False):
+    return _convert_fx(graph_module, inplace, debug, is_dynamic=True)
+
+def _quantize_fx(model, qconfig_dict, run_fn=None, run_args=None, inplace=False, debug=False, is_dynamic=False):
```

Comment:
can we use the enum instead of `is_dynamic`, it's more readable

Reply:
replied in the previous comment, it looks like we do not need to distinguish between static and qat

Reply:
`is_dynamic` can mean many things, `QuantType.DYNAMIC` is very explicit.  Can we use the enum?  It's valuable to have a single enum where we can branch between dynamic, static, QAT, even if not every single piece of code uses all the three branches.

Reply:
I mean we don't branch based on static or qat, the only thing we are checking in the code is whether it is dynamic quantization or not, so defining three quant types where two of them are not used seems redundant

Reply:
I can rename is_dynamic to is_dynamic_quantization to make it more explicit

Reply:
@jerryzh168 , I was just reading through the implementation of `torch/quantization/fx/quantize.py`, and there is a branch for QAT logic, see 

```
if input_root.training:
    self._qat_swap_modules(input_root)
```

can we keep the enum we already have, and switch this line to use it?  There are 3 quant types, with 3 branches in the code.  It's also not ideal to use `root.training` silently, it's not intuitive for this to represent "this is QAT".

Reply:
we already used training in the implementation of fusion: https://github.com/pytorch/pytorch/blob/master/torch/quantization/fuse_modules.py#L11, I think it's better to keep consistent. actually I think this branch is not needed, since qat_swap_module will have no effect when we call it on eval models.